tactics/tactics.ml

```diff
@@ -3038,8 +3050,13 @@ let specialize (c,lbind) ipat =
         | Context.Rel.Declaration.LocalAssum _::lp' , t::l' ->
           let sigma,hd' = rebuild_lambdas sigma lp' (args@[t]) hd l' in
           sigma,hd'
+        | Context.Rel.Declaration.LocalDef _::lp' , _ ->
+           (* letins are have been reduced in l and should anyway not
+              correspond to an arg, we ignore them. *)
+          let sigma,hd' = rebuild_lambdas sigma lp' args hd l in
+          sigma,hd'
```

Comment:
This is the real code change. Others are mostly comments and one simplification (thd vs c_hd which are the same).